test/t/index/test_dump_and_load_index.cpp

```diff
@@ -0,0 +1,45 @@
+#include "catch.hpp"
+
+#include <osmium/index/detail/tmpfile.hpp>
+#include <osmium/index/map/dense_file_array.hpp>
+#include <osmium/index/map/dense_mmap_array.hpp>
+#include <osmium/osm/location.hpp>
+#include <osmium/osm/types.hpp>
+
+using dense_mmap_array_t = osmium::index::map::DenseMmapArray<osmium::unsigned_object_id_type, osmium::Location>;
+using dense_file_array_t = osmium::index::map::DenseFileArray<osmium::unsigned_object_id_type, osmium::Location>;
+
+TEST_CASE("Dump DenseMmapArray, load as DenseFileArray") {
+    const int fd = osmium::detail::create_tmp_file();
+    REQUIRE(osmium::file_size(fd) == 0);
+    constexpr const size_t S = sizeof(dense_file_array_t::element_type);
```

Comment:
`S` is too short a name.

Reply:
I chose a longer name in uppercase (because it's a constant) but please have a look at https://github.com/osmcode/libosmium/blob/master/test/t/index/test_file_based_index.cpp#L24 ;)